### PR TITLE
Added `destroy` config to make auto-destroying a container optional

### PR DESCRIPTION
**Pull request by btoll**
This will be helpful when using agents that will be used to create
snapshots of our package repos.

After all, switching out existing snapshots created from a previous run
of the container is much faster than publishing a snapshot from a new
local cache.

### CODE REVIEW

File: mule/task/docker/__init__.py [hunk 1]
```diff
@@ -118,17 +113,19 @@ def pullFromDockerHub(self, image):
             found = True
         return found
 
-    def run(self, image, command, work_dir, volumes, env):
+    def run(self, image):
```
Comment (btoll) [inline mule/task/docker/__init__.py:116]: This function signature was beginning to get unwieldy.

**Review by onetechnical — APPROVED** [review]
I haven't run it but these changes look fine to me.

**Review by egieseke — APPROVED** [review]
Looks ok.  

Food for thought, I wonder if we can simplify our docker interaction using the docker SDK. 
https://docker-py.readthedocs.io/en/stable/

**Comment by btoll** [comment]
> Looks ok.
> 
> Food for thought, I wonder if we can simplify our docker interaction using the docker SDK.
> https://docker-py.readthedocs.io/en/stable/

Probably be worth looking into.

**Pull request merged**